### PR TITLE
[F] Finish implementing Facebook SSO

### LINKED ISSUE #25: Finalize authentication in client

### PR DESCRIPTION
**Pull request by dananjohnson**
Resolves #25 

## What this change does ##

This PR adds the missing dialog which sends a mutation to the API to register and authenticate a user signing in with their Facebook account.

## Notes for reviewers ##

Note that this feature won't fully work if the Facebook app isn't successfully authenticating you and redirecting you back to `localhost`. This was the case for me, but after going back and forth with Blake it seems like the issue is with my local or with me not being an admin on the Facebook app. He was able to go through this process just fine. See the thread in Slack, near the bottom, for the discussion.

- 5 = "Please make sure there are no obvious errors and that you believe it does what it says it does"

## Testing ##

Make sure to update your API repo to use `http://localhost:3000/?facebook_sso=true` for the Facebook redirect URL.

## Checklist ##

If any of the following are true, please check them off
- [ ] This is a breaking change: changes in page data (`/src/data/pages/PAGE.json`) or scientific data (anything in `/static/`) and I will notify Product Marketing when it is in prod and user visible.
- [ ] This change impacts several investigations (e.g. the change affects reuseed styles, widgets, pages, QAs, utility methods, etc.)
- [X] This change is isolated to a specific page or Component (i.e. it's a one-off)
- [ ] I don't know what this change does

### CODE REVIEW

File: components/auth/dialogs/SignInFacebook/SignInFacebook.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        return t("facebook_sso.loading");
+      default:
+        return undefined;
+    }
+  }
+
+  function getDescription() {
+    switch (status) {
+      case "success":
+        return t("facebook_sso.success_message", { context: pendingGroup });
```
Comment (alexgoff) [inline components/auth/dialogs/SignInFacebook/SignInFacebook.tsx:70]: there's a context here for "pendingGroup", but I don't see a matching key in the translation strings, is it missing or is this context unused?

**Review by alexgoff — CHANGES_REQUESTED** [review]
just one small note to check that an i18n string has a matching context, otherwise looks ok

File: public/localeStrings/en/translation.json [hunk 2]
```diff
@@ -243,6 +243,16 @@
     "loading": "Activating your account…",
     "confirm_button": "Got it!"
   },
+  "facebook_sso": {
+    "success": "Success!",
+    "success_message": "Your account has been activated. Please log in to begin using the site as a registered user.",
+    "success_message_educator": "Your account has been activated, but it will need to be approved by an administrator before you can access educator content. You will receive an email when you have been approved.",
```
Comment (dananjohnson) [inline public/localeStrings/en/translation.json:249]: @alexgoff here's the key that uses the `pendingGroup` context, falling back to just `success_message` as the default.

Comment (dananjohnson) [reply public/localeStrings/en/translation.json:249]: See https://www.i18next.com/translation-function/context

Comment (alexgoff) [reply public/localeStrings/en/translation.json:249]: oh, I didn't catch that `pendingGroup` was a variable and not a string, I was looking for a key like "success_message_pendingGroup" 🤯  everything looks good 😓

**Review by alexgoff — APPROVED** [review]

**Pull request merged**